qutip/core/data/convert.pyx

```diff
@@ -259,14 +261,26 @@ cdef class _to:
                     _converter(convert[::-1], to_t, from_t)
         for dispatcher in self.dispatchers:
             dispatcher.rebuild_lookup()
+        for dtype in self.dtypes:
+            self._str2type[dtype.__name__.lower()] = dtype
+
+    def parse(self, dtype):
+        if isinstance(dtype, type):
```

Comment:
Also, second thought - since this is the fast-path, I think it might be a good idea to have the first test be `if type(dtype) is type` instead.  We know that the only thing we store in our `dict` are immediate `type` instances and never subclasses, so we know for certain it'll fail if the passed argument is a subclass and `type(x) is type` is the fastest possible test.